test/integ/user_comments_integ_test.js

```diff
@@ -0,0 +1,278 @@
+                    mimeType: "image/jpeg",
+                },
+            ], TEST_USER);
+            // Write comment on image
+            const writeResult = await writeComment(agent, TEST_USER, uploadedImages[0].id, COMMENT_TEXT);
+            if (writeResult.statusCode !== 200) {
+                assert.fail(`Could not write comment, status code: ${writeResult.statusCode}, resp: ${JSON.stringify(writeResult.body)}`);
+            }
+            // Verify that image link exists in API response
+            const commentsResult = await getUserComments(agent, TEST_USER, "html");
+            assert.equal(commentsResult.statusCode, 200);
+            const commentsBody = commentsResult.text;
+            assert.match(commentsBody, new RegExp(`"/images/${uploadedImages[0].id}"`, "g"));
+            assert.match(commentsBody, new RegExp(`"/images/${uploadedImages[0].id}.jpeg"`, "g"));
+        });
+
+        it("should return posted timestamp within the comment HTML", async () => {
+            // Upload an image
+            const uploadedImages = await addImagesForUser([
+                {
+                    fileName: "Black_tea_pot_cropped.jpg",
+                    mimeType: "image/jpeg",
+                },
+            ], TEST_USER);
+            // Write comment on image
+            const writeResult = await writeComment(agent, TEST_USER, uploadedImages[0].id, COMMENT_TEXT);
+            if (writeResult.statusCode !== 200) {
+                assert.fail(`Could not write comment, status code: ${writeResult.statusCode}, resp: ${JSON.stringify(writeResult.body)}`);
+            }
+            // Get posted comment from response
+            const postedComment = writeResult.body.message;
+            // Verify that posted date exists in API response
+            const commentsResult = await getUserComments(agent, TEST_USER, "html");
+            assert.equal(commentsResult.statusCode, 200);
+            const commentsBody = commentsResult.text;
+            assert.match(commentsBody, new RegExp(new Date(postedComment.postedDate).toUTCString(), "g"));
+        });
+
+        it("should return removed image link within comment HTML if referenced image has been deleted", async () => {
+            // Upload an image
+            const uploadedImages = await addImagesForUser([
+                {
+                    fileName: "Black_tea_pot_cropped.jpg",
+                    mimeType: "image/jpeg",
+                },
+            ], TEST_USER);
+            // Write comment on image
+            const writeResult = await writeComment(agent, TEST_USER, uploadedImages[0].id, COMMENT_TEXT);
+            if (writeResult.statusCode !== 200) {
+                assert.fail(`Could not write comment, status code: ${writeResult.statusCode}, resp: ${JSON.stringify(writeResult.body)}`);
+            }
+            // Delete image
+            const deleteResult = await deleteImage(agent, uploadedImages[0].id);
```

Comment:
Consider creating helper functions to perform action and assert it succeeded.

Reply:
Will take care of this when I clean and consolidate the testing helper functions.